e2e/testcases/main_test.go

```diff
@@ -19,33 +19,56 @@ import (
 	"fmt"
 	"math/rand"
 	"os"
+	"strconv"
 	"testing"
 	"time"
 
 	"kpt.dev/configsync/e2e"
 	"kpt.dev/configsync/e2e/nomostest"
 )
 
+// This is a bit of a hack to enforce our --parallel flag over the --test.parallel
+// parameter of `go test`. The `go test` defaults to GOMAXPROCS, which is sane
+// for unit tests but is an undesirable default for e2e tests which create
+// a cluster per thread. Default to 1 instead.
+func setParallelFlag() error {
+	err := flag.Set("test.parallel", strconv.Itoa(e2e.NumParallel()))
+	return err
+}
+
 func TestMain(m *testing.M) {
+	os.Exit(main(m))
```

Comment:
Might be cleaner to have main return an error and handle printing it and exiting 1 or 0.

Reply:
oh, i see m.Run returns a code... bleh.
I suppose you could make an error type that has an error code field and type check it.

Reply:
I would generally agree, however the `m.Run()` method returns an exit code so I think returning an exit code is cleaner in this context